File: client/src/app/models/order.ts
export interface Order {
    id: number
    buyerEmail: string
    shippingAddress: ShippingAddress
    orderDate: string
    orderItems: OrderItem[]
    discount: number
    total: number
    orderStatus: string
}

export interface ShippingAddress {
    name: string
    line1: string
    line2?: string | null
    city: string
    state: string
    postal_code: string
    country: string
}

export interface OrderItem {
    productId: number
    name: string
    pictureUrl: string
    price: number
    quantity: number
}

export interface CreateOrder {
    shippingAddress: ShippingAddress
}

File: API/Entities/OrderAgregate/Order.cs
namespace API.Entities.OrderAgregate;

public class Order
{
    public int Id { get; set; }
    public required string BuyerEmail { get; set; }
    public required ShippingAddress ShippingAddress { get; set; }
    public DateTime OrderDate { get; set; } = DateTime.UtcNow;
    public List<OrderItem> OrderItems { get; set; } = [];
    public OrderStatus OrderStatus { get; set; } =  OrderStatus.Pending;
}